dGame/dComponents/TriggerComponent.cpp

```diff
@@ -0,0 +1,249 @@
+#include "TriggerComponent.h"
+
+#include "dZoneManager.h"
+#include "LUTriggers.h"
+
+#include "MissionComponent.h"
+#include "PhantomPhysicsComponent.h"
+
+#include "CDMissionTasksTable.h"
+
+
+TriggerComponent::TriggerComponent(Entity* parent, const std::string triggerInfo): Component(parent) {
+	m_Parent = parent;
+	m_Trigger = nullptr;
+
+	std::stringstream ss(triggerInfo);
+	std::vector<std::string> tokens;
+	std::string token;
+	while (std::getline(ss, token, ':')) {
+		tokens.push_back(token);
+	}
+
+	uint32_t sceneID = std::stoi(tokens[0]);
+	uint32_t triggerID = std::stoi(tokens[1]);
+
+	if (m_Trigger) {
+		delete m_Trigger;
+		m_Trigger = nullptr;
+	}
+
+	m_Trigger = dZoneManager::Instance()->GetZone()->GetTrigger(sceneID, triggerID);
+
+	if (!m_Trigger) m_Trigger = new LUTriggers::Trigger();
+}
+
+void TriggerComponent::TriggerEvent(std::string eventID, Entity* optionalTarget) {
+	if (m_Trigger != nullptr && m_Trigger->enabled) {
+		for (LUTriggers::Event* triggerEvent : m_Trigger->events) {
+			if (triggerEvent->eventID == eventID) {
+				for (LUTriggers::Command* cmd : triggerEvent->commands) {
+					HandleTriggerCommand(cmd->id, cmd->target, cmd->targetName, cmd->args, optionalTarget);
+				}
+			}
+		}
+	}
+}
+
+void TriggerComponent::HandleTriggerCommand(std::string id, std::string target, std::string targetName, std::string args, Entity* optionalTarget) {
+	auto argArray = ParseArgs(args);
+
+	// determine targets
+	std::vector<Entity*> targetEntities;
+	if (target == "objGroup") targetEntities = EntityManager::Instance()->GetEntitiesInGroup(targetName);
+	if (target == "self") targetEntities.push_back(m_Parent);
+	if (optionalTarget) targetEntities.push_back(optionalTarget);
+
+	// if we have no targets, then we are done
+	if (targetEntities.empty()) return;
+
+	for (Entity* targetEntity : targetEntities) {
+		if (!targetEntity) continue;
+
+		if (id == "SetPhysicsVolumeEffect") {
```

Comment:
DLU sure does love its long if-else chains.

Reply:
same, hate it but it works. Very much open to suggestions and I'll do some searches to see if anything better pops up

Reply:
TL;DR everything else is so much uglier in code with no real benefit